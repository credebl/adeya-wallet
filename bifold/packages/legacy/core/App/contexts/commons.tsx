import React, { createContext, useContext } from 'react'

import { AppConsoleLogger, LogLevel } from '../services/logger'

export interface AppCommonsContext {
  log: (message: string, logLevel: Exclude<LogLevel, LogLevel.off>) => void
}

export const AppCommons = createContext<AppCommonsContext>(null as unknown as AppCommonsContext)

export const CommonUtilProvider: React.FC = ({ children }) => {
  const logger = new AppConsoleLogger(LogLevel.test)
  const log = async (messaage: string, logLevel: Exclude<LogLevel, LogLevel.off>) => {
    logger.log(logLevel, messaage)
    //TODO: do some more logic like collecting errors for later logging and investigation
  }

  return (
    <AppCommons.Provider
      value={{
        log: log,
      }}
    >
      {children}
    </AppCommons.Provider>
  )
}

export const useCommons = () => {
  const commonsContext = useContext(AppCommons)
  if (!commonsContext) {
    throw new Error('commonsContext must be used within a CommonUtilProvider')
  }
  return commonsContext
}
