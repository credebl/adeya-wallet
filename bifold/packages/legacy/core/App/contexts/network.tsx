import { NetInfoStateType, useNetInfo } from '@react-native-community/netinfo'
import * as React from 'react'
import { createContext, useContext, useState } from 'react'

import NetInfoModal from '../components/modals/NetInfoModal'
import { fetchLedgerNodes, canConnectToLedgerNode } from '../utils/ledger'

export interface NetworkContext {
  silentAssertConnectedNetwork: () => boolean
  assertConnectedNetwork: () => boolean
  displayNetInfoModal: () => void
  hideNetInfoModal: () => void
  assertLedgerConnectivity: () => Promise<boolean>
}

export const NetworkContext = createContext<NetworkContext>(null as unknown as NetworkContext)

export const NetworkProvider: React.FC = ({ children }) => {
  const netInfo = useNetInfo()
  const [isNetInfoModalDisplayed, setIsNetInfoModalDisplayed] = useState<boolean>(false)

  const displayNetInfoModal = () => {
    setIsNetInfoModalDisplayed(true)
  }

  const hideNetInfoModal = () => {
    setIsNetInfoModalDisplayed(false)
  }

  const silentAssertConnectedNetwork = () => {
    return netInfo.isConnected || netInfo.type !== NetInfoStateType.none
  }

  const assertConnectedNetwork = () => {
    const isConnected = silentAssertConnectedNetwork()
    if (!isConnected) {
      displayNetInfoModal()
    }

    return isConnected
  }

  const assertLedgerConnectivity = async (): Promise<boolean> => {
    const nodes = fetchLedgerNodes()

    if (typeof nodes === 'undefined' || nodes.length === 0) {
      return false
    }

    const connections = await Promise.all(nodes.map((n: { host: string; port: number }) => canConnectToLedgerNode(n)))
    return connections.includes(true)
  }

  return (
    <NetworkContext.Provider
      value={{
        silentAssertConnectedNetwork,
        assertConnectedNetwork,
        displayNetInfoModal,
        hideNetInfoModal,
        assertLedgerConnectivity,
      }}
    >
      {children}
      <NetInfoModal visible={isNetInfoModalDisplayed} onSubmit={() => hideNetInfoModal()} />
    </NetworkContext.Provider>
  )
}

export const useNetwork = () => useContext(NetworkContext)
