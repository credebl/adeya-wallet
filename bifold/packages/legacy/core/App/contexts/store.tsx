import React, { createContext, Dispatch, useContext, useReducer } from 'react'

import { State } from '../types/state'

import _defaultReducer, { ReducerAction } from './reducers/store'

type Reducer = <S extends State>(state: S, action: ReducerAction<any>) => S

interface StoreProviderProps {
  initialState?: State
  reducer?: Reducer
}

export const defaultState: State = {
  onboarding: {
    didAgreeToTerms: false,
    didCompleteTutorial: false,
    didCreatePIN: false,
    didConsiderBiometry: false,
  },
  authentication: {
    didAuthenticate: false,
  },
  loginAttempt: {
    loginAttempts: 0,
    servedPenalty: true,
  },
  lockout: {
    displayNotification: false,
  },
  preferences: {
    developerModeEnabled: false,
    biometryPreferencesUpdated: false,
    useBiometry: false,
    useVerifierCapability: false,
    useConnectionInviterCapability: false,
  },
  tours: {
    seenToursPrompt: false,
    enableTours: true,
    seenHomeTour: false,
  },
  deepLink: {
    activeDeepLink: '',
  },
  loading: false,
}

export const StoreContext = createContext<[State, Dispatch<ReducerAction<any>>]>([
  defaultState,
  () => {
    return
  },
])

export const mergeReducers = (a: Reducer, b: Reducer): Reducer => {
  return <S extends State>(state: S, action: ReducerAction<any>): S => {
    return a(b(state, action), action)
  }
}

export const defaultReducer = _defaultReducer

export const StoreProvider: React.FC<StoreProviderProps> = ({ children, initialState, reducer }) => {
  const _reducer = reducer ?? defaultReducer
  const _state = initialState ?? defaultState
  const [state, dispatch] = useReducer(_reducer, _state)

  return <StoreContext.Provider value={[state, dispatch]}>{children}</StoreContext.Provider>
}

export const useStore = <S extends State>(): [S, Dispatch<ReducerAction<any>>] => {
  const context = useContext(StoreContext)

  return context as unknown as [S, Dispatch<ReducerAction<any>>]
}
