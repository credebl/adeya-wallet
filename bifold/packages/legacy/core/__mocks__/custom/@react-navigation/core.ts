const navigate = jest.fn()

const navigation = {
  navigate,
  setOptions: jest.fn(),
  getParent: jest.fn(() => ({
    navigate,
  })),
  getState: jest.fn(() => ({
    index: jest.fn(),
  })),
  goBack: jest.fn(),
  pop: jest.fn(),
  reset: jest.fn(),
}

const useNavigation = () => {
  return navigation
}

const useIsFocused = () => {
  return true
}

export { useNavigation, useIsFocused }
