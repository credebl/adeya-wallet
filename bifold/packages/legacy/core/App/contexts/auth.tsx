import { agentDependencies } from '@aries-framework/react-native'
import React, { createContext, useContext, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { DeviceEventEmitter } from 'react-native'

import { EventTypes } from '../constants'
import { DispatchAction } from '../contexts/reducers/store'
import { useStore } from '../contexts/store'
import {
  secretForPIN,
  storeWalletSecret,
  loadWalletSecret,
  loadWalletSalt,
  isBiometricsActive,
  wipeWalletKey,
} from '../services/keychain'
import { WalletSecret } from '../types/security'
import { hashPIN } from '../utils/crypto'

export interface AuthContext {
  checkPIN: (PIN: string) => Promise<boolean>
  getWalletCredentials: () => Promise<WalletSecret | undefined>
  removeSavedWalletSecret: () => void
  disableBiometrics: () => Promise<void>
  setPIN: (PIN: string) => Promise<void>
  commitPIN: (useBiometry: boolean) => Promise<boolean>
  isBiometricsActive: () => Promise<boolean>
}

export const AuthContext = createContext<AuthContext>(null as unknown as AuthContext)

export const AuthProvider: React.FC = ({ children }) => {
  const [walletSecret, setWalletSecret] = useState<WalletSecret>()
  const [, dispatch] = useStore()
  const { t } = useTranslation()

  const setPIN = async (PIN: string): Promise<void> => {
    const secret = await secretForPIN(PIN)
    await storeWalletSecret(secret)
  }

  const getWalletCredentials = async (): Promise<WalletSecret | undefined> => {
    if (walletSecret && walletSecret.key) {
      return walletSecret
    }

    const { secret, err } = await loadWalletSecret(
      t('Biometry.UnlockPromptTitle'),
      t('Biometry.UnlockPromptDescription')
    )

    DeviceEventEmitter.emit(EventTypes.BIOMETRY_ERROR, err !== undefined)

    if (!secret) {
      return
    }
    setWalletSecret(secret)

    return secret
  }

  const commitPIN = async (useBiometry: boolean): Promise<boolean> => {
    const secret = await getWalletCredentials()
    if (!secret) {
      return false
    }
    // set did authenticate to true if we can get wallet credentials
    dispatch({
      type: DispatchAction.DID_AUTHENTICATE,
    })
    if (useBiometry) {
      await storeWalletSecret(secret, useBiometry)
    } else {
      // erase wallet key if biometrics is disabled
      await wipeWalletKey(useBiometry)
    }
    return true
  }

  const checkPIN = async (PIN: string): Promise<boolean> => {
    const secret = await loadWalletSalt()

    if (!secret || !secret.salt) {
      return false
    }

    const hash = await hashPIN(PIN, secret.salt)

    try {
      await agentDependencies.indy.openWallet({ id: secret.id }, { key: hash })
      // need full secret in volatile memory in case user wants to fall back to using PIN
      const fullSecret = await secretForPIN(PIN, secret.salt)
      setWalletSecret(fullSecret)
      return true
    } catch (e) {
      return false
    }
  }

  const removeSavedWalletSecret = () => {
    setWalletSecret(undefined)
  }

  const disableBiometrics = async () => {
    await wipeWalletKey(true)
  }

  return (
    <AuthContext.Provider
      value={{
        checkPIN,
        getWalletCredentials,
        removeSavedWalletSecret,
        disableBiometrics,
        commitPIN,
        setPIN,
        isBiometricsActive,
      }}
    >
      {children}
    </AuthContext.Provider>
  )
}

export const useAuth = () => useContext(AuthContext)
