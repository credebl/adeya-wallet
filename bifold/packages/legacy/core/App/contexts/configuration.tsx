import { IndyPoolConfig } from '@aries-framework/core'
import { createContext, ReducerAction, useContext } from 'react'

import { ProofRequestTemplate } from '../../verifier'
import { EmptyListProps } from '../components/misc/EmptyList'
import { RecordProps } from '../components/record/Record'
import OnboardingPages from '../screens/OnboardingPages'
import { ScanProps } from '../screens/Scan'
import { OCABundleResolver } from '../types/oca'
import { PINSecurityParams } from '../types/security'
import { SettingSection } from '../types/settings'

interface NotificationConfiguration {
  component: React.FC
  onCloseAction: (dispatch?: React.Dispatch<ReducerAction<any>>) => void
  title: string
  description: string
  buttonTitle: string
  pageTitle: string
}

export interface ConfigurationContext {
  pages: typeof OnboardingPages
  splash: React.FC
  terms: React.FC
  homeContentView: React.FC
  credentialListHeaderRight: React.FC
  credentialListOptions: React.FC
  credentialEmptyList: React.FC<EmptyListProps>
  developer: React.FC
  OCABundleResolver: OCABundleResolver
  scan: React.FC<ScanProps>
  useBiometry: React.FC
  record: React.FC<RecordProps>
  PINSecurity: PINSecurityParams
  indyLedgers: IndyPoolConfig[]
  settings: SettingSection[]
  customNotification: NotificationConfiguration
  useCustomNotifications: () => { total: number; notifications: any }
  connectionTimerDelay?: number
  autoRedirectConnectionToHome?: boolean
  proofRequestTemplates?: Array<ProofRequestTemplate>
  enableTours?: boolean
}

export const ConfigurationContext = createContext<ConfigurationContext>(null as unknown as ConfigurationContext)

export const ConfigurationProvider = ConfigurationContext.Provider

export const useConfiguration = () => useContext(ConfigurationContext)
