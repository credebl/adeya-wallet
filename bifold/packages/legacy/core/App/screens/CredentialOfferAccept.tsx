import { CredentialState } from '@aries-framework/core'
import { useCredentialById } from '@aries-framework/react-hooks'
import { useNavigation } from '@react-navigation/core'
import React, { useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { Platform, Modal, StatusBar, StyleSheet, Text, View, ScrollView } from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'

import Button, { ButtonType } from '../components/buttons/Button'
import { useAnimatedComponents } from '../contexts/animated-components'
import { useConfiguration } from '../contexts/configuration'
import { useTheme } from '../contexts/theme'
import { Screens, TabStacks } from '../types/navigators'
import { statusBarStyleForColor, StatusBarStyles } from '../utils/luminance'
import { testIdWithKey } from '../utils/testable'

enum DeliveryStatus {
  Pending,
  Completed,
  Declined,
}

export interface CredentialOfferAcceptProps {
  visible: boolean
  credentialId: string
}

const CredentialOfferAccept: React.FC<CredentialOfferAcceptProps> = ({ visible, credentialId }) => {
  const { t } = useTranslation()
  const [shouldShowDelayMessage, setShouldShowDelayMessage] = useState<boolean>(false)
  const [credentialDeliveryStatus, setCredentialDeliveryStatus] = useState<DeliveryStatus>(DeliveryStatus.Pending)
  const [timerDidFire, setTimerDidFire] = useState<boolean>(false)
  const [timer, setTimer] = useState<NodeJS.Timeout>()
  const credential = useCredentialById(credentialId)
  const navigation = useNavigation()
  const { ListItems } = useTheme()
  const { CredentialAdded, CredentialPending } = useAnimatedComponents()
  const { connectionTimerDelay } = useConfiguration()
  const connTimerDelay = connectionTimerDelay ?? 10000 // in ms
  const styles = StyleSheet.create({
    container: {
      ...ListItems.credentialOfferBackground,
      height: '100%',
      padding: 20,
    },
    image: {
      marginTop: 20,
    },
    messageContainer: {
      alignItems: 'center',
    },
    messageText: {
      fontWeight: 'normal',
      textAlign: 'center',
      marginTop: 30,
    },
    controlsContainer: {
      marginTop: 'auto',
      margin: 20,
    },
    delayMessageText: {
      textAlign: 'center',
      marginTop: 20,
    },
  })

  if (!credential) {
    throw new Error('Unable to fetch credential from AFJ')
  }

  const onBackToHomeTouched = () => {
    navigation.getParent()?.navigate(TabStacks.HomeStack, { screen: Screens.Home })
  }

  const onDoneTouched = () => {
    navigation.getParent()?.navigate(TabStacks.CredentialStack, { screen: Screens.Credentials })
  }

  useEffect(() => {
    if (credential.state === CredentialState.CredentialReceived || credential.state === CredentialState.Done) {
      timer && clearTimeout(timer)
      setCredentialDeliveryStatus(DeliveryStatus.Completed)
    }
  }, [credential])

  useEffect(() => {
    if (timerDidFire || credentialDeliveryStatus !== DeliveryStatus.Pending || !visible) {
      return
    }

    const timer = setTimeout(() => {
      setShouldShowDelayMessage(true)
      setTimerDidFire(true)
    }, connTimerDelay)

    setTimer(timer)

    return () => {
      timer && clearTimeout(timer)
    }
  }, [visible])

  return (
    <Modal visible={visible} transparent={true} animationType={'none'}>
      <SafeAreaView style={{ ...ListItems.credentialOfferBackground }}>
        <StatusBar
          barStyle={
            Platform.OS === 'android' ? StatusBarStyles.Light : statusBarStyleForColor(styles.container.backgroundColor)
          }
        />
        <ScrollView style={[styles.container]}>
          <View style={[styles.messageContainer]}>
            {credentialDeliveryStatus === DeliveryStatus.Pending && (
              <Text
                style={[ListItems.credentialOfferTitle, styles.messageText]}
                testID={testIdWithKey('CredentialOnTheWay')}
              >
                {t('CredentialOffer.CredentialOnTheWay')}
              </Text>
            )}

            {credentialDeliveryStatus === DeliveryStatus.Completed && (
              <Text
                style={[ListItems.credentialOfferTitle, styles.messageText]}
                testID={testIdWithKey('CredentialAddedToYourWallet')}
              >
                {t('CredentialOffer.CredentialAddedToYourWallet')}
              </Text>
            )}
          </View>

          <View style={[styles.image, { minHeight: 250, alignItems: 'center', justifyContent: 'flex-end' }]}>
            {credentialDeliveryStatus === DeliveryStatus.Completed && <CredentialAdded />}
            {credentialDeliveryStatus === DeliveryStatus.Pending && <CredentialPending />}
          </View>

          {shouldShowDelayMessage && credentialDeliveryStatus === DeliveryStatus.Pending && (
            <Text
              style={[ListItems.credentialOfferDetails, styles.delayMessageText]}
              testID={testIdWithKey('TakingTooLong')}
            >
              {t('Connection.TakingTooLong')}
            </Text>
          )}
        </ScrollView>

        <View style={[styles.controlsContainer]}>
          {credentialDeliveryStatus === DeliveryStatus.Pending && (
            <View>
              <Button
                title={t('Loading.BackToHome')}
                accessibilityLabel={t('Loading.BackToHome')}
                testID={testIdWithKey('BackToHome')}
                onPress={onBackToHomeTouched}
                buttonType={ButtonType.ModalSecondary}
              />
            </View>
          )}

          {credentialDeliveryStatus === DeliveryStatus.Completed && (
            <View>
              <Button
                title={t('Global.Done')}
                accessibilityLabel={t('Global.Done')}
                testID={testIdWithKey('Done')}
                onPress={onDoneTouched}
                buttonType={ButtonType.ModalPrimary}
              />
            </View>
          )}
        </View>
      </SafeAreaView>
    </Modal>
  )
}

export default CredentialOfferAccept
