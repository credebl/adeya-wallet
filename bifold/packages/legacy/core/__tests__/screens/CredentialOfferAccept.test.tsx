import { CredentialExchangeRecord as CredentialRecord, CredentialState } from '@aries-framework/core'
import { useCredentialById } from '@aries-framework/react-hooks'
import { render, waitFor } from '@testing-library/react-native'
import fs from 'fs'
import path from 'path'
import React from 'react'

import { ConfigurationContext } from '../../App/contexts/configuration'
import CredentialOfferAccept from '../../App/screens/CredentialOfferAccept'
import { testIdWithKey } from '../../App/utils/testable'
import configurationContext from '../contexts/configuration'
import timeTravel from '../helpers/timetravel'

jest.mock('@react-navigation/core', () => {
  return require('../../__mocks__/custom/@react-navigation/core')
})
jest.mock('@react-navigation/native', () => {
  return require('../../__mocks__/custom/@react-navigation/native')
})

const credentialId = '123'
const credentialPath = path.join(__dirname, '../fixtures/degree-credential.json')
const credential = JSON.parse(fs.readFileSync(credentialPath, 'utf8'))
const credentialRecord = new CredentialRecord(credential)
// TODO:(jl) Make a fn to revive JSON dates properly and pass to `parse`
credentialRecord.createdAt = new Date(credentialRecord.createdAt)

// @ts-ignore
useCredentialById.mockReturnValue(credentialRecord)

describe('displays a credential accept screen', () => {
  test('renders correctly', () => {
    const tree = render(
      <ConfigurationContext.Provider value={configurationContext}>
        <CredentialOfferAccept visible={true} credentialId={credentialId} />
      </ConfigurationContext.Provider>
    )

    const doneButton = tree.queryByTestId('Done')

    expect(tree).toMatchSnapshot()
    expect(doneButton).toBeNull()
    expect(useCredentialById).toBeCalledWith(credentialId)
  })

  test('transitions to taking too delay message', async () => {
    const tree = render(
      <ConfigurationContext.Provider value={configurationContext}>
        <CredentialOfferAccept visible={true} credentialId={credentialId} />
      </ConfigurationContext.Provider>
    )

    await waitFor(() => {
      timeTravel(11000)
    })

    const backToHomeButton = tree.getByTestId(testIdWithKey('BackToHome'))
    const doneButton = tree.queryByTestId(testIdWithKey('Done'))

    expect(tree).toMatchSnapshot()
    expect(backToHomeButton).not.toBeNull()
    expect(doneButton).toBeNull()
  })

  test('transitions to offer accepted', () => {
    credentialRecord.state = CredentialState.CredentialReceived

    const tree = render(
      <ConfigurationContext.Provider value={configurationContext}>
        <CredentialOfferAccept visible={true} credentialId={credentialId} />
      </ConfigurationContext.Provider>
    )

    const doneButton = tree.getByTestId(testIdWithKey('Done'))
    const backToHomeButton = tree.queryByTestId(testIdWithKey('BackToHome'))

    expect(tree).toMatchSnapshot()
    expect(doneButton).not.toBeNull()
    expect(backToHomeButton).toBeNull()
  })
})
