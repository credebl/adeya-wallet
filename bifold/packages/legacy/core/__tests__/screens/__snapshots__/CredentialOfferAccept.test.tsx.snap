// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays a credential accept screen renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <RNCSafeAreaView
    style={
      Object {
        "backgroundColor": "#000000",
      }
    }
  >
    <RCTScrollView
      style={
        Array [
          Object {
            "backgroundColor": "#000000",
            "height": "100%",
            "padding": 20,
          },
        ]
      }
    >
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 26,
                  "fontWeight": "bold",
                },
                Object {
                  "fontWeight": "normal",
                  "marginTop": 30,
                  "textAlign": "center",
                },
              ]
            }
            testID="com.ariesbifold:id/CredentialOnTheWay"
          >
            CredentialOffer.CredentialOnTheWay
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "marginTop": 20,
              },
              Object {
                "alignItems": "center",
                "justifyContent": "flex-end",
                "minHeight": 250,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "column",
                },
              ]
            }
          >
            <
              height={110}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "marginTop": -30,
                    "position": "absolute",
                  },
                ]
              }
              width={110}
            />
            <View
              style={
                Object {
                  "opacity": 0,
                  "transform": Array [
                    Object {
                      "translateY": -90,
                    },
                  ],
                }
              }
            >
              <
                height={110}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "marginLeft": 10,
                      "position": "absolute",
                    },
                  ]
                }
                width={110}
              />
            </View>
            <
              height={140}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
              width={140}
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Array [
          Object {
            "margin": 20,
            "marginTop": "auto",
          },
        ]
      }
    >
      <View>
        <View
          accessibilityLabel="Loading.BackToHome"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderColor": "#42803E",
              "borderRadius": 4,
              "borderWidth": 2,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="com.ariesbifold:id/BackToHome"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#42803E",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              Loading.BackToHome
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</Modal>
`;

exports[`displays a credential accept screen transitions to offer accepted 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <RNCSafeAreaView
    style={
      Object {
        "backgroundColor": "#000000",
      }
    }
  >
    <RCTScrollView
      style={
        Array [
          Object {
            "backgroundColor": "#000000",
            "height": "100%",
            "padding": 20,
          },
        ]
      }
    >
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 26,
                  "fontWeight": "bold",
                },
                Object {
                  "fontWeight": "normal",
                  "marginTop": 30,
                  "textAlign": "center",
                },
              ]
            }
            testID="com.ariesbifold:id/CredentialAddedToYourWallet"
          >
            CredentialOffer.CredentialAddedToYourWallet
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "marginTop": 20,
              },
              Object {
                "alignItems": "center",
                "justifyContent": "flex-end",
                "minHeight": 250,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "column",
                },
              ]
            }
          >
            <View
              collapsable={false}
              nativeID="animatedComponent"
              style={
                Object {
                  "alignItems": "center",
                  "marginBottom": 10,
                  "opacity": 0,
                }
              }
            >
              <
                height={45}
                width={45}
              />
            </View>
            <View>
              <
                height={110}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "marginTop": -30,
                      "position": "absolute",
                    },
                  ]
                }
                width={110}
              />
              <View
                collapsable={false}
                nativeID="animatedComponent"
                style={
                  Object {
                    "opacity": 0,
                    "transform": Array [
                      Object {
                        "translateY": -90,
                      },
                    ],
                  }
                }
              >
                <
                  height={110}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "marginLeft": 10,
                        "position": "absolute",
                      },
                    ]
                  }
                  width={110}
                />
              </View>
              <
                height={140}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ]
                }
                width={140}
              />
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Array [
          Object {
            "margin": 20,
            "marginTop": "auto",
          },
        ]
      }
    >
      <View>
        <View
          accessibilityLabel="Global.Done"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#42803E",
              "borderRadius": 4,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="com.ariesbifold:id/Done"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              Global.Done
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</Modal>
`;

exports[`displays a credential accept screen transitions to taking too delay message 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <RNCSafeAreaView
    style={
      Object {
        "backgroundColor": "#000000",
      }
    }
  >
    <RCTScrollView
      style={
        Array [
          Object {
            "backgroundColor": "#000000",
            "height": "100%",
            "padding": 20,
          },
        ]
      }
    >
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 26,
                  "fontWeight": "bold",
                },
                Object {
                  "fontWeight": "normal",
                  "marginTop": 30,
                  "textAlign": "center",
                },
              ]
            }
            testID="com.ariesbifold:id/CredentialOnTheWay"
          >
            CredentialOffer.CredentialOnTheWay
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "marginTop": 20,
              },
              Object {
                "alignItems": "center",
                "justifyContent": "flex-end",
                "minHeight": 250,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "column",
                },
              ]
            }
          >
            <
              height={110}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "marginTop": -30,
                    "position": "absolute",
                  },
                ]
              }
              width={110}
            />
            <View
              style={
                Object {
                  "opacity": 1,
                  "transform": Array [
                    Object {
                      "translateY": -16.252128433746066,
                    },
                  ],
                }
              }
            >
              <
                height={110}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "marginLeft": 10,
                      "position": "absolute",
                    },
                  ]
                }
                width={110}
              />
            </View>
            <
              height={140}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
              width={140}
            />
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              Object {
                "marginTop": 20,
                "textAlign": "center",
              },
            ]
          }
          testID="com.ariesbifold:id/TakingTooLong"
        >
          Connection.TakingTooLong
        </Text>
      </View>
    </RCTScrollView>
    <View
      style={
        Array [
          Object {
            "margin": 20,
            "marginTop": "auto",
          },
        ]
      }
    >
      <View>
        <View
          accessibilityLabel="Loading.BackToHome"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderColor": "#42803E",
              "borderRadius": 4,
              "borderWidth": 2,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="com.ariesbifold:id/BackToHome"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#42803E",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              Loading.BackToHome
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</Modal>
`;
